Log and bail when there are dir creation issues (SOFTWARE-5531)

brianhlin opened the pull request:
Includes #169 . I've opted to just go with the exception route -- there aren't that many call sites:

1. https://github.com/opensciencegrid/gratia-probe/blob/2.x/common/gratia/common/send.py#L305-L305
2. https://github.com/opensciencegrid/gratia-probe/blob/2.x/common/gratia/common/send.py#L91-L91

But `Send` has quite a bit of call sites so I think I prefer just bailing loudly with an exception in this case because if the working dir isn't writeable then there's going to be a lot of trouble.

djw8605 committed:
Add logging when there are failures to create sandbox directories

This can happen if directory ownership is incorrect.

djw8605 committed:
Update common/gratia/common/sandbox_mgmt.py

Co-authored-by: Brian Lin <brianhlin@gmail.com>

djw8605 committed:
Update common/gratia/common/sandbox_mgmt.py

Co-authored-by: Brian Lin <brianhlin@gmail.com>

brianhlin committed:
Remove duplicate traceback

brianhlin committed:
Stop processing if there are working dir issues (SOFTWARE-5531)

matyasselmeci commented on common/gratia/common/sandbox_mgmt.py:
```diff
@@ -569,21 +569,29 @@ def OpenNewRecordFile(dirIndex):
-                except:
-                    continue
+                except Exception as exc:
+                    msg = 'Warning: Exception caught while creating directory: ' + working_dir
```
```suggestion
                    msg = 'Fatal: Exception caught while creating directory: ' + working_dir
```
if you're bailing, it's not a warning.

brianhlin replied:
Good point

brianhlin committed:
Clarify that we're blowing things up in the message

brianhlin commented:
Since we're giving ourselves a bit more time on this release, I want to just try/except the write. marking `do-not-merge`

brianhlin committed:
Simplify error processing for dir creation (SOFTWARE-5531)

1. If the dir doesn't exist, we'll have thrown an exception
2. If the dir isn't writable, exceptions will be raised by subsequent
   write attempts to the dir

brianhlin committed:
Rip out custom mkdir code (SOFTWARE-5531)

brianhlin commented:
@matyasselmeci it felt good to get rid of that. I'm planning on adding a linting GHA just to doubly make sure we're not shooting ourselves in the foot. I can include that in this PR or a separate one.

matyasselmeci reviewed (approved):
yay!
